lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/JobLifeCycleManager.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Timers;
+using Org.Apache.REEF.Driver;
+using Org.Apache.REEF.IMRU.API;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Utilities.Logging;
+using Org.Apache.REEF.IMRU.OnREEF.Parameters;
+
+namespace Org.Apache.REEF.IMRU.OnREEF.Driver
+{
+    /// <summary>
+    /// JobLifecycleManager orchestrates job cancellation flow.
+    /// If job cancellation detector is configured in job definition, the manager starts a timer and periodically checks for cancellation signal.
+    /// if cancellation signal is detected, the manager creates JobCancelled event and propagates the event to all subscriber.
+    /// the manager is used by IMRU driver to enable job cancellation based on job definition.
+    /// </summary>
+    internal sealed class JobLifeCycleManager :
+        IDisposable,
+        IJobLifecycleManager,
+        IObserver<IDriverStarted>
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(JobLifeCycleManager));
+
+        private Timer timer;
```

Comment:
fixed (that's why I like using properties instead :-)

---
In reply to: [86014940](https://github.com/apache/reef/pull/1161#discussion_r86014940) [](ancestors = 86014940)